### DISCUSSION TITLE
self-hosted runners in organization don't show up in public repositories

### DISCUSSION THREAD

**Discussion by dormant-user** [community/community#118477] [Actions] [answered]
### Select Topic Area

Bug

### Body

I have 5 idle runners in my organization https://github.com/thevickypedia, but none of my public repositories are able to share these runners.
I see a banner saying, `You don't have any self-hosted runners for this repository`
I'm able to see those runners in my private repo, and jobs are getting picked instantly.
I've verified my organization settings, and everything is set to share runners with all repositories
I even tried to disable and re-enable them, still no luck.

<details>
<summary>Organization settings for runners</summary>

![Screenshot 2024-04-09 at 7 27 57 PM](https://github.com/community/community/assets/166179991/a410ee4b-d9fb-4c86-b1bb-41624b287773)
</details>

Sample public repos where runners don't show up
- https://github.com/thevickypedia/rustream/actions/runners?tab=self-hosted
- https://github.com/thevickypedia/vpn-server/actions/runners?tab=self-hosted
- https://github.com/thevickypedia/expose/actions/runners?tab=self-hosted

<details>
<summary>Private repo where runners are available</summary>

![Screenshot 2024-04-09 at 7 31 07 PM](https://github.com/community/community/assets/166179991/a7057944-8f0f-43ce-bbfc-ac32bbbdbc54)
</details>

> 📓 This is the private repo where I originally had repository level `self-hosted` runners, before removing them and converting my personal profile into an organization. I'm not sure if that's the reason why this one got easily synced up and others are kinda ghosted.

I created a ticket with GitHub support team, but also wanted to check with the community to see if anyone else is facing similar issues.
Any help or guidance would be much appreciated. Thank you.

**Comment by KittyChiu** [comment, marked as answer]
Have a look if there is existing runner group restricts the access.  For runner group, it has an option to disable usage by public repos.
![image](https://github.com/community/community/assets/42864823/c2c43b78-be9c-4ad7-ae6c-368e7e921e63)

For security hardening, it is [recommended](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#hardening-for-self-hosted-runners) that self-hosted runners should not be used by public repos.

**Reply by dormant-user** [reply]
You're right, that's where I have the restriction for public repos. Thanks for your help, I'll keep the security concerns in mind and make appropriate changes before enabling for selected public repos.
Once again thank you so much.
